zcash_client_sqlite/src/wallet/init/migrations/orchard_shardtree.rs

```diff
@@ -0,0 +1,214 @@
+                shard_index,
+                start_position,
+                end_position_exclusive,
+                subtree_start_height,
+                subtree_end_height,
+                contains_marked;",
+        )?;
+
+        // Treat the current best-known chain tip height as the height to use for Orchard
+        // initialization, bounded below by NU5 activation.
+        if let Some(orchard_init_height) = scan_queue_extrema(transaction)?.and_then(|r| {
+            self.params
+                .activation_height(NetworkUpgrade::Nu5)
+                .map(|orchard_activation| std::cmp::max(orchard_activation, *r.end()))
+        }) {
+            // If a scan range exists that contains the Orchard init height, split it in two at the
+            // init height.
+            if let Some((start, end, range_priority)) = transaction
+                .query_row_and_then(
+                    "SELECT block_range_start, block_range_end, priority
+                     FROM scan_queue
+                     WHERE block_range_start <= :orchard_init_height
+                     AND block_range_end > :orchard_init_height",
+                    named_params![":orchard_init_height": u32::from(orchard_init_height)],
+                    |row| {
+                        let start = BlockHeight::from(row.get::<_, u32>(0)?);
+                        let end = BlockHeight::from(row.get::<_, u32>(1)?);
+                        let range_priority: i64 = row.get(2)?;
+                        Ok((start, end, range_priority))
+                    },
+                )
+                .optional()?
+            {
+                transaction.execute(
+                    "DELETE from scan_queue WHERE block_range_start = :start",
+                    named_params![":start": u32::from(start)],
+                )?;
+                // Rewrite the start of the scan range to be exactly what it was prior to the
+                // change.
+                transaction.execute(
+                    "INSERT INTO scan_queue (block_range_start, block_range_end, priority)
+                     VALUES (:block_range_start, :block_range_end, :priority)",
+                    named_params![
+                        ":block_range_start": u32::from(start),
+                        ":block_range_end": u32::from(orchard_init_height),
+                        ":priority": range_priority,
+                    ],
+                )?;
+                // Rewrite the remainder of the range to have priority `Historic`
+                transaction.execute(
+                    "INSERT INTO scan_queue (block_range_start, block_range_end, priority)
+                     VALUES (:block_range_start, :block_range_end, :priority)",
+                    named_params![
+                        ":block_range_start": u32::from(orchard_init_height),
+                        ":block_range_end": u32::from(end),
+                        ":priority": priority_code(&ScanPriority::Historic),
```

Comment:
This should not decrease the existing priority (as the range we select could have any priority):
```suggestion
                        ":priority": priority_code(&range_priority.max(ScanPriority::Historic)),
```
(or whatever syntax works).